### ISSUE TITLE
Mixed keybindings after switching keyboard layout on Linux

### ISSUE THREAD

**Issue by chrmarti** [Microsoft/vscode#23690]
- VSCode Version: Code - Insiders 1.11.0-insider (6b9548065ad3e31fca4f0e91cf882505fe8c30cf, 2017-03-30T06:18:34.366Z)
- OS Version: Linux x64 3.13.0-108-generic
- Extensions: none
---

Found verifying #1549

Steps to Reproduce on Linux under Parallels with a US keyboard (did not reproduce on Mac):

1. Start with US layout
2. Navigate in code to populate navigation history
3. Navigate back / forward with keyboard shortcut (Ctrl + Alt + Minus, Ctrl + Shift + Minus)
4. Switch to German layout
5. Navigate back / forward now works with both the German location of Minus (expected) and the US location of Minus (unexpected)
6. Reload window
7. Now navigation only works with the German location of Minus (expected)

It seems like the layout switch doesn't clear out the old binding.

**Comment by alexandrudima** [comment]
Yes, keyboard layout changes are not detected on Linux. The workaround is to restart vscode

**Comment by nicolaslussier** [comment]
restarting VS code dos not make it loar right keybord for shortkut

**Comment by alexandrudima** [comment]
@nicolaslussier Have you tried the workaround suggested at https://github.com/Microsoft/vscode/issues/24166#issuecomment-294002885

**Comment by Spone** [comment]
I'm having trouble with this, especially because I open VS Code with the French keyboard layout enabled (my OS default), then I switch to US for coding. The `Ctrl+Z` and `Ctrl+W` bindings stay on the French keyboard layout which is super frustrating... I haven't found a workaround that helps for now.

**Comment by alexandrudima** [comment]
@Spone You can reload VS Code `F1 > Developer: Reload Window`

**Comment by Spone** [comment]
Yes, but doing so every time I launch VS Code? :rofl:

**Comment by alexandrudima** [comment]
More possible workarounds:
- launch `code` with a script where you change the keyboard layout right before launching via `setxkbmap`

(or)

- use keyboard layout independent keybindings (code based). e.g. use the following format in your `keybindings.json` for those keys that annoy you:
```
{ "key": "cmd+[KeyZ]", "command": "undo", "when": "textInputFocus && !editorReadonly" },
```

**Comment by Spone** [comment]
Oh, great! The second workaround did the job! Thanks a lot @alexandrudima that was the main pain point while adopting VS Code :)

**Comment by lodi-g** [comment]
Coming from #88269 .

If it matters, my keyboard layout is set via `localectl set-x11-keymap` and thus should be correct when X is starting. 

X starts my WM, my WM starts VSCode, shouldn't it work?

Thanks

**Comment by thibauld** [comment]
I am switching frequently between EN and FR keyboard. This bug makes VSCode barely usable for me on Ubuntu 18.04.

**Comment by GabrielTouron** [comment]
I am using EN and FR keyboards on Ubuntu 18.04. I made a custom alias for the command : `setxkbmap fr`  that I use when I'm switching keyboard. It works for me.
Thank you @alexdima

**Comment by jerlam06** [comment]
Exact same issue, cannot find a way to fix.

**Comment by jeantil** [comment]
@alexdima you mention 
> use keyboard layout independent keybindings (code based). e.g. use the following format in your keybindings.json for those keys that annoy you:
> { "key": "cmd+[KeyZ]", "command": "undo", "when": "textInputFocus && !editorReadonly" },

Since this is not how the default keybindings are defined, I assume that there are drawbacks to this method, can you tell what they are (or why the default keybindings don't use this ? )

thanks

**Comment by achertovsky** [comment]
Same issue here. Nothing changed, except update to Version: 1.53.2

**Comment by fonji** [comment]
> I would just prefer to have all the open workspaces to be open again automatically.

File > exit (or `Ctrl + Q`) will quit all windows
Then when you start `code` again (without any parameter), it restores the latest windows by default.

**Comment by pironim** [comment]
ubuntu 16 and latest vs code still an issue.
